patches/minecraft/net/minecraft/block/WoodType.java.patch

```diff
@@ -0,0 +1,39 @@
+--- a/net/minecraft/block/WoodType.java
++++ b/net/minecraft/block/WoodType.java
+@@ -18,6 +18,7 @@
+ 
+    protected WoodType(String p_i225775_1_) {
+       this.field_227045_h_ = p_i225775_1_;
++      this.namespace = "minecraft";
+    }
+ 
+    private static WoodType func_227047_a_(WoodType p_227047_0_) {
+@@ -32,4 +33,28 @@
+    public String func_227048_b_() {
+       return this.field_227045_h_;
+    }
++
++   private static final Set<WoodType> MOD_VALUES = java.util.Collections.newSetFromMap(new java.util.concurrent.ConcurrentHashMap<>());
```

Comment:
All of this is complicated and unneded.
The constructor takes a string.
Leave the name field as is, add a new RL field and a getter
There is also no need to track 'mod values' people can call register if they want it to be in the values list.